src/lib/graveyard-container.js

```diff
@@ -0,0 +1,19 @@
+class Graveyard {
```

Comment:
Having a lib/graveyard is great if you want to split out some static functions like "get home for object".  In that case I'd suggest having the lib/graveyard be a static class with only static methods, and moving the refObject version to the object-specific script.

Reply:
Thinking on it you could also have a static "move object to home" method for even cleaner use.

Reply:
I've kept the instantiated class that's attached to the actual game object that is the graveyard bag. It makes sense to me to have a class instance associated with the actual game object. But all it does for now is attach the handler to the container ref and call the `ReturnGameObjectHome` utility class that contains the actual logic.

Reply:
Cool, `patch-unit-bag.js` looks good - so for units all we need to do is find the right player bag then call `addObjects` on it, nice.

Have also had a look at `r-swap-split-combine.js` as you suggested. There's a nice way we can make finding certain things really easily - how about adding a custom object to the world object to hold references to anything we may want to grab quickly, like a player's unit bag? We could add it during setup and it might look something like the below. Then we create utility functions that we can call from anywhere to grab the correct thing from our custom object.

```js
world.__ti4 = {
  players: {
    player1: {
        player: {{playerRef}},
        unitBags: {
          infantry: {{containerRef}},
          fighters: {{containerRef}},
          dreadnoughts: {{containerRef}},
          ..etc
        },
        reinforcements: {{objectRef}},
        ....etc
    },
    ....etc
  },
  graveyardBag: {{containerRef}},
  decks: {
    action: {
      deck: {{objectRef}},
      discard: {{objectRef}},
      mat: {{objectRef}}
    },
    ....etc
  }
}

const getPlayerUnitBag = (player, unitType) => {
  return world.__ti4.players[player.id].unitBags[unitType]
}
```

Reply:
```js
const { set } = require('lodash')

const addPlayerUnitBagReference = (player, unitType, ref) => {
  set(world, `__ti4.players.${player.id}.unitBags.${unitType}`, ref)
}
```

Reply:
That's an option.  I'm a fan of static objects that can fix things behind the scenes vs a "table".  For instance, maybe a player deletes something and spawns a new one in.  Something could be watching the onObjectDestroyed and onObjectCreated events but it's hard to get all the corner cases right.

If instead there were a `GraveyardBag.get()` static method that returned a GameObject reference, it could internally keep a cached reference, and internally do the "oops the ref is gone, find it again and update the cache" recovery.

Reply:
Yeah true, keeping those globally-cached references up-to-date could get tricky for the edge-cases, although I wasn't thinking that code would access the references directly. Rather, a bunch of utility functions such as `getPlayerUnitBag` would lookup the references from the cache, and could deal with any issues behind the scenes, similar to what you're talking about with a `GraveyardBag.get()`, only global.

Either way, I'll create those utility functions for use by the graveyward bag or anything else that might need them and fill them with similar logic to what we currently use to find game objects. I'll leave the "global reference cache" idea with you. Let me know if you think it'd be worthwhile to add in.

Reply:
Great, sounds like we're agreed on helper functions to find things.  I'd say go ahead and do it for the graveyard bag and we can migrate things to a more global location if/when needed.